Compiler/src/AST.hs

```diff
@@ -38,8 +38,17 @@ data Transformer = Call String [Locator]
                  | Construct String [Locator]
     deriving (Show, Eq)
 
+data Op = OpLt | OpGt | OpLe | OpGe | OpEq | OpNe | OpIn
+    deriving (Show, Eq)
+
+data Precondition = Conj [Precondition]
+                  | BinOp Op Locator Locator
```

Comment:
Should preconditions be limited to binary operators? For example, it might be useful to use `!` in a precondition.